### ISSUE TITLE
OSError: illegal IP address string passed to inet_pton

### ISSUE THREAD

**Issue by dennisjenkins75** [timmo001/system-bridge#2845]
### Description

I installed systembridge onto two Ubuntu PCs yesterday, and successfully integrated them into Home Assistant (I want to be able to remotely power them down).  Excellent.  That worked.

However, today, the systembridge.service crashes while starting up (on both systems).

```
-- Boot a5f1d9dc2ee04a37bb6c8c46a781a9d6 --
Nov 10 11:43:06 artemis systemd[1]: Started System Bridge.
Nov 10 11:43:10 artemis systembridge[1205]: 2023-11-10 11:43:10 INFO (MainThread) [systembridgebackend.server.api] Serving frontend from: /usr/local/share/systembridge/_internal/systembridgefrontend/out
Nov 10 11:43:10 artemis systembridge[1205]: 2023-11-10 11:43:10 INFO (MainThread) [systembridgebackend.Application] System Bridge 3.10.0: Startup
Nov 10 11:43:10 artemis systembridge[1205]: 2023-11-10 11:43:10 INFO (MainThread) [systembridgebackend.Application] Your api-key is: **REDACTED**
Nov 10 11:43:10 artemis systembridge[1205]: 2023-11-10 11:43:10 INFO (MainThread) [systembridgebackend.Application] Autostart enabled: True
Nov 10 11:43:12 artemis systembridge[1205]: 2023-11-10 11:43:12 CRITICAL (MainThread) [root] Unhandled error in application
Nov 10 11:43:12 artemis systembridge[1205]: Traceback (most recent call last):
Nov 10 11:43:12 artemis systembridge[1205]:   File "systembridge.py", line 51, in application
Nov 10 11:43:12 artemis systembridge[1205]:   File "systembridgebackend/__init__.py", line 57, in __init__
Nov 10 11:43:12 artemis systembridge[1205]:     self._server = Server(
Nov 10 11:43:12 artemis systembridge[1205]:                    ^^^^^^^
Nov 10 11:43:12 artemis systembridge[1205]:   File "systembridgebackend/server/__init__.py", line 72, in __init__
Nov 10 11:43:12 artemis systembridge[1205]:     self._mdns_advertisement.advertise_server()
Nov 10 11:43:12 artemis systembridge[1205]:   File "systembridgebackend/server/mdns.py", line 42, in advertise_server
Nov 10 11:43:12 artemis systembridge[1205]:     info = ServiceInfo(
Nov 10 11:43:12 artemis systembridge[1205]:            ^^^^^^^^^^^^
Nov 10 11:43:12 artemis systembridge[1205]:   File "src/zeroconf/_services/info.py", line 189, in zeroconf._services.info.ServiceInfo.__init__
Nov 10 11:43:12 artemis systembridge[1205]:   File "zeroconf/_utils/net.py", line 70, in _encode_address
Nov 10 11:43:12 artemis systembridge[1205]: OSError: illegal IP address string passed to inet_pton
```

Both PCs are on the same LAN and use DHCP to acquire their IP address, and both run a local docker instance.

```
root@artemis:~# ifconfig | egrep "(^[a-z]|inet)"
docker0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 172.17.255.255
enp5s0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.0.4  netmask 255.255.255.0  broadcast 192.168.0.255
        inet6 fd8b:6647:f36a::f25  prefixlen 128  scopeid 0x0<global>
        inet6 fd8b:6647:f36a:0:271d:ac84:120c:6b8d  prefixlen 64  scopeid 0x0<global>
        inet6 fd8b:6647:f36a:0:beb8:54f3:13ab:e007  prefixlen 64  scopeid 0x0<global>
        inet6 fe80::32b1:7311:dee4:ddc4  prefixlen 64  scopeid 0x20<link>
enp6s0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
```

When I restart the failed service with `systemctl restart systembridge.service`, after the machine has its IP address, it runs fine.  However, I'd like it to work properly while a DHCP address is not yet available.

I assume that I can just change the systemd `Service.Restart` option to something else though?

```
root@artemis:~# cat /etc/systemd/system/systembridge.service 
[Unit]
Description=System Bridge
After=network.target

[Service]
ExecStart=/usr/bin/systembridge --no-gui
Restart=on-abort
RemainAfterExit=yes

[Install]
WantedBy=multi-user.target
```

### What Platform / OS are you running?

Linux

### What version are you running?

3.10.0

### Anything in the logs or a references that might be useful?

```txt
Nov 10 11:43:12 artemis systembridge[1205]: Traceback (most recent call last):
Nov 10 11:43:12 artemis systembridge[1205]:   File "systembridge.py", line 51, in application
Nov 10 11:43:12 artemis systembridge[1205]:   File "systembridgebackend/__init__.py", line 57, in __init__
Nov 10 11:43:12 artemis systembridge[1205]:     self._server = Server(
Nov 10 11:43:12 artemis systembridge[1205]:                    ^^^^^^^
Nov 10 11:43:12 artemis systembridge[1205]:   File "systembridgebackend/server/__init__.py", line 72, in __init__
Nov 10 11:43:12 artemis systembridge[1205]:     self._mdns_advertisement.advertise_server()
Nov 10 11:43:12 artemis systembridge[1205]:   File "systembridgebackend/server/mdns.py", line 42, in advertise_server
Nov 10 11:43:12 artemis systembridge[1205]:     info = ServiceInfo(
Nov 10 11:43:12 artemis systembridge[1205]:            ^^^^^^^^^^^^
Nov 10 11:43:12 artemis systembridge[1205]:   File "src/zeroconf/_services/info.py", line 189, in zeroconf._services.info.ServiceInfo.__init__
Nov 10 11:43:12 artemis systembridge[1205]:   File "zeroconf/_utils/net.py", line 70, in _encode_address
Nov 10 11:43:12 artemis systembridge[1205]: OSError: illegal IP address string passed to inet_pton
```

**Comment by funforums** [comment]
Hey there, I am not sure if I got your goal.
I do get that you would like to prevent System Bridge to fail, but if it does not have an IP, I suppose it cannot be reachable, so what would be your goal of having it running anyway? Do you have in mind that it should try to re-bind itself later, in the hope the machine has acquired an address? Or is it that you'd rather express a dependency to DHCP in your service, to make sure it does not fail? 
Cheers

**Comment by dennisjenkins75** [comment]
Hello.  I suppose that ti depends on if systembridge it attempting to bind "all addresses", or does something else.  I have no special config telling it to bind a specific address, or to a specific NIC.  I did not study the systembridge code yet, as it is written in python and I prefer C++.

I originally configured the systemd.service file per the online docs, but after posting this issue, I modified it slightly to just auto-restart if it crashes.  That addressed my problem, but it might not be elegant.  And posting it here might help someone else.

```
root@artemis:~# cat /etc/systemd/system/systembridge.service 
[Unit]
Description=System Bridge
After=network.target

[Service]
ExecStart=/usr/bin/systembridge --no-gui
Restart=always
RemainAfterExit=no

[Install]
WantedBy=multi-user.target
```
